### PR TITLE
Add Playwright to XFD

### PR DESCRIPTION
**Pull request by JCantu248**
#  #

## 🗣 Description ##

Add Playwright tests to the Crossfeed repository. 
## 💭 Motivation and context ##

Adds automated frontend UI tests to the project. 
## 🧪 Testing ##

Test cases were created and tested against Crossfeed in its current state. All UI tests are passing and working in the intended manner. 

run `cd frontend/xfd_playwright`
then run `npx playwright install`
finally run `npx playwright test`

- [x] Revert dependencies to default branches.
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by nickviola — APPROVED** [review]
LGTM

**Comment by schmelz21** [comment]
@jsf9k , @dav3r - Can your team take a look at this one. We are introducing Playwright into our tests. This initial commit is to  establish the toolset and initial functionality. Follow-up PRs we will be issued to gain better coverage on UI tests.

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+name: Playwright Tests
+on:
+  deployment_status:
+    paths:
+      - 'playwright/**'
+      - '.github/workflows/playwright.yml'
```
Comment (jsf9k) [inline .github/workflows/playwright.yml]: I don't think these quotes are necessary.
```suggestion
      - playwright/**
      - .github/workflows/playwright.yml
```

**Review by jsf9k — APPROVED** [review]
I suggest removing some unnecessary quotes, but otherwise this looks fine to me.  (With the caveat that I don't know the first thing about Playwright.)

File: playwright/.gitignore [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+/playwright/.cache/
+/test-results/
+/playwright-report/
+/blob-report/
+/playwright/.cache/
+/test-results/
+/playwright-report/
+/blob-report/
+/playwright/.cache/
+storageState.json
```
Comment (dav3r) [inline playwright/.gitignore]: Sort lines alphabetically and remove duplicated entries:
```suggestion
/blob-report/
/playwright-report/
/playwright/.cache/
/test-results/
node_modules/
storageState.json
```

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+defaults:
+  run:
+    working-directory: ./playwright
+env:
+  PW_XFD_URL: ${{ vars.PW_XFD_URL }}
+  PW_XFD_USERNAME: ${{ secrets.PW_XFD_USERNAME }}
+  PW_XFD_PASSWORD: ${{ secrets.PW_XFD_PASSWORD }}
+  PW_XFD_2FA_ISSUER: ${{ secrets._PW_XFD_2FA_ISSUER }}
+  PW_XFD_2FA_SECRET: ${{ secrets.PW_XFD_2FA_SECRET }}
+  PW_XFD_USER_ROLE: ${{ vars.PW_XFD_USER_ROLE }}
```
Comment (dav3r) [inline .github/workflows/playwright.yml]: Sort environment variables alphabetically:
```suggestion
  PW_XFD_2FA_ISSUER: ${{ secrets._PW_XFD_2FA_ISSUER }}
  PW_XFD_2FA_SECRET: ${{ secrets.PW_XFD_2FA_SECRET }}
  PW_XFD_PASSWORD: ${{ secrets.PW_XFD_PASSWORD }}
  PW_XFD_URL: ${{ vars.PW_XFD_URL }}
  PW_XFD_USER_ROLE: ${{ vars.PW_XFD_USER_ROLE }}
  PW_XFD_USERNAME: ${{ secrets.PW_XFD_USERNAME }}
```

**Review by dav3r — APPROVED** [review]
LGTM aside from the two minor things I noted.

**Comment by dav3r** [comment]
It should also be noted that this PR has failing vulnerability checks that should be resolved.  I don't know if those are the same as mentioned in #127, but regardless, it's always preferable to have all checks pass in a PR.

Yes -These are being addressed currently in that ticket and have a high prioritization. This ticket is updated and would then force us to have clean checks each PR moving forward.

**Pull request merged**